### PR TITLE
#2533 - update tables with headers

### PR DESCRIPTION
**Pull request by samrichca**
# Description of changes

- add feature flag for v8 tables due to missing features (resizing, width/maxWidth)
- update these tables:
    - Samples from this Specimen (Clinical Timeline, Donor Entity)
    - Treatments (Clinical Timeline, Donor Entity)
    - Available Files (Donor Entity)
    - My Memberships (Profile)

**Requires Uikit update** https://github.com/icgc-argo/uikit/pull/107

## Type of Change

- [ ] Bug
- [x] Dependency updates
- [ ] Feature
- [x] Refactoring
- [ ] Release candidate
- [ ] Styling
- [ ] Testing
- [ ] Other (please describe)

## Checklist before requesting review

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [x] No changes to UI
- Back-end (select one):
  - [ ] Back-end PRs merged, tested on develop, and linked in this PR
  - [x] No back-end changes
- [x] Manually tested changes
- [x] Added copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/donor-entity/ClinicalTimeline/index.tsx [hunk 1]
```diff
@@ -120,11 +120,8 @@ const renderSelectedDataRow = (selectedData, selectedSamples) => {
 const ClinicalTimeline = ({ data }: { data: DonorCentricRecord }) => {
   // TODO: Remove test values
   const entityData = formatTimelineEntityData(data);
-  const entities = [
-    entityData.primary_diagnosis,
-    ...entityData.specimens,
-    ...mockTimelineData.slice(1),
-  ];
+  const entities = [entityData.primary_diagnosis, ...entityData.specimens, ...mockTimelineData];
```
Comment (samrichca) [inline components/pages/donor-entity/ClinicalTimeline/index.tsx:123]: this restores the "Treatments" tab (dev/QA only) in Clinical Timeline

File: components/pages/user/ProgramAccessBox/index.tsx [hunk 2]
```diff
@@ -54,6 +57,18 @@ interface Column {
 
 const ProgramTable = (props: { programs: Array<T_ProgramTableProgram> }) => {
   const { permissions } = useAuthContext();
+  const ProgramNameCellOld = ({ original }: { original: T_ProgramTableProgram }) => (
```
Comment (samrichca) [inline components/pages/user/ProgramAccessBox/index.tsx:60]: I wonder if this was removed in a recent PR. I didn't add it

File: components/pages/user/ProgramAccessBox/index.tsx [hunk 3]
```diff
@@ -78,17 +95,54 @@ const ProgramTable = (props: { programs: Array<T_ProgramTableProgram> }) => {
...
+              ? PROGRAMS_LIST_PATH
+              : PROGRAM_DASHBOARD_PATH.replace(PROGRAM_SHORT_NAME_PATH, info.getValue())
+          }
+        >
+          <Link>{info.getValue()}</Link>
+        </NextLink>
+      ),
+      header: () => 'Program Name',
+      id: 'shortName',
+      // maxWidth: 150,
```
Comment (samrichca) [inline components/pages/user/ProgramAccessBox/index.tsx:127]: haven't implemented width yet

File: components/pages/user/ProgramAccessBox/index.tsx [hunk 4]
```diff
@@ -97,15 +151,26 @@ const ProgramTable = (props: { programs: Array<T_ProgramTableProgram> }) => {
...
-      <Table
-        withOutsideBorder
-        parentRef={containerRef}
-        sortable={false}
-        showPagination={false}
-        data={props.programs}
-        columns={allCols}
-        getTdProps={(_, row, column) => ({ style: { whiteSpace: 'normal' } })}
-      />
+      {FEATURE_REACT_TABLE_V8_ENABLED ? (
```
Comment (caravinci) [inline components/pages/user/ProgramAccessBox/index.tsx:154]: clever! may do the same for VirusSeq

File: components/pages/user/ProgramAccessBox/index.tsx [hunk 5]
```diff
@@ -78,17 +95,54 @@ const ProgramTable = (props: { programs: Array<T_ProgramTableProgram> }) => {
...
     ? [
-        ...baseCols.slice(0, 1),
+        ...baseColumns.slice(0, 1),
         {
           Header: 'Membership Type',
           accessor: 'membershipType',
           Cell: (props) => capitalize(props.value),
         },
-        ...baseCols.slice(1),
+        ...baseColumns.slice(1),
```
Comment (caravinci) [inline components/pages/user/ProgramAccessBox/index.tsx:106]: thank you!!

File: components/pages/user/ProgramAccessBox/index.tsx [hunk 6]
```diff
@@ -65,9 +80,11 @@ const ProgramTable = (props: { programs: Array<T_ProgramTableProgram> }) => {
       <Link>{original.shortName}</Link>
     </NextLink>
   );
+
   const containerRef = createRef<HTMLDivElement>();
 
-  const baseCols: Column[] = [
+  // react table v6
+  const baseColumns: Column[] = [
```
Comment (caravinci) [inline components/pages/user/ProgramAccessBox/index.tsx:87]: thank you! those few extra letters feel so much better

File: components/pages/donor-entity/ClinicalTimeline/index.tsx (hunk 1 above)
Comment (caravinci) [reply components/pages/donor-entity/ClinicalTimeline/index.tsx:123]: May be worth making a comment of some sort to indicate that this `slice` does that (I, or a new team member would never guess that)

File: components/pages/user/ProgramAccessBox/index.tsx (hunk 2 above)
Comment (samrichca) [reply components/pages/user/ProgramAccessBox/index.tsx]: removed it

File: components/pages/donor-entity/ClinicalTimeline/index.tsx (hunk 1 above)
Comment (samrichca) [reply components/pages/donor-entity/ClinicalTimeline/index.tsx:123]: I'll add a comment if BAs want me to add the slice back, for sure

**Review by caravinci — APPROVED** [review]

**Review by joneubank — APPROVED** [review]

**Pull request merged**